src/Psalm/Internal/Analyzer/Statements/Expression/CastAnalyzer.php

```diff
@@ -328,11 +323,7 @@ public static function castIntAttempt(
 
         $atomic_types = $stmt_type->getAtomicTypes();
 
-        $parent_nodes = [];
-
-        if ($statements_analyzer->data_flow_graph instanceof VariableUseGraph) {
```

Comment:
what is the reason behind this removal?

Reply:
We need to set `$parent_nodes` to `$stmt_type->parent_nodes` in the case `$statements_analyzer->data_flow_graph` is an instance of `VariableUseGraph` (previously needed) or an instance of `TaintedFlowGraph` (for this bugfix).

Since `VariableUseGraph` and `TaintedFlowGraph` are the 2 only classes that extend `DataFlowGraph`, `$stmt_type->parent_nodes` will always fulfill the condition. Therefore, the `if` statement is useless and we can remove it.